planner/core/rule_join_elimination.go

```diff
@@ -122,7 +122,7 @@ func (o *outerJoinEliminator) isInnerJoinKeysContainUniqueKey(innerPlan LogicalP
 // check whether one of index sets is contained by inner join index
 func (o *outerJoinEliminator) isInnerJoinKeysContainIndex(innerPlan LogicalPlan, joinKeys *expression.Schema) (bool, error) {
 	ds, ok := innerPlan.(*DataSource)
-	if !ok {
+	if !ok || joinKeys.Len() == 0 {
```

Comment:
Should we check the `joinKeys.Len() == 0 && len(otherConditions) > 0`?

Reply:
Why we need to check `otherConditions`? In this case, len(otherConditions) == 0.